src/db_adapters/coin/legacy/mod.rs

```diff
@@ -13,7 +13,12 @@ pub(crate) async fn collect_legacy(
     shard_id: &near_indexer_primitives::types::ShardId,
     receipt_execution_outcomes: &[near_indexer_primitives::IndexerExecutionOutcomeWithReceipt],
     block_header: &near_indexer_primitives::views::BlockHeaderView,
+    chain_id: &str,
 ) -> anyhow::Result<Vec<CoinEvent>> {
+    // We don't need to store legacy events for testnet
+    if chain_id == "testnet" {
```

Comment:
@telezhnaya I would consider inverting the logic: "we don't need to store legacy events for non-mainnet networks" (be it testnet, betanet, shardnet, etc)